cmd/run-function/runfn/runfn.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package runfn
+//Reference: sigs.k8s.io/kustomize/kyaml/runfn
+package runfn
+
+import (
+	"io"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+
+	"path/filepath"
+	"sigs.k8s.io/kustomize/kyaml/fn/runtime/container"
+	"sigs.k8s.io/kustomize/kyaml/fn/runtime/runtimeutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// RunFns runs the set of configuration functions in a local directory against
+// the Resources in that directory
+type RunFns struct {
+	// Path is the path to the directory containing resources to run the functions on
+	Path string
+
+	// Function is a function to run against the input.
+	Function *runtimeutil.FunctionSpec
+
+	// FnConfig is the configurations passed from command line
+	FnConfig *yaml.RNode
+
+	// Output can be set to write the result to Output
+	Output io.Writer
+
+	// Env contains environment variables that will be exported to container
+	Env []string
+
+	// functionFilterProvider provides a filter to perform the function.
+	// this is a variable, so it can be mocked in tests
+	functionFilterProvider func(filter runtimeutil.FunctionSpec, fnConfig *yaml.RNode) (kio.Filter, error)
+}
```

Comment:
What I meant was, does something break if you `import` their `runfn` package without using all of the `RunFn` struct members?

My personal preference would be to import wherever possible, if we must fork in order to support `podman` then we should do that and look to contribute upstream.

Reply:
I've not tried using their RunFns, but I don't think it will break anything, we need to adjust how we pass in the required configuration. I'll try using import






> My personal preference would be to import wherever possible, if we must fork in order to support `podman` then we should do that and look to contribute upstream.

Do you mean adding a PR to kustomize?

Reply:
> Do you mean adding a PR to kustomize?

yup